### PR TITLE
Partially complete add/edit ingredient to recipe

### LINKED ISSUE #21: US 02.02.01
# User Story
As a meal planner, I want to add an ingredient to a recipe, with a brief description, amount, unit, and ingredient category.

# Acceptance Criteria
- Given
  - A possible way to click on a recipe object already inputted by the user.
- Action
  - Once clicked should bring up a fragment that allows the user to view the data stored, and a way to add ingredients.
  - Once clicked, bring up a fragment that allows data input for ingredient objects.
- Then
  - Store user input data in the recipe object.
- Additional components
  - Storyboard demonstrating the feature
  - Wiki entry describing feature
  - UML for frontend components.

### PR DESCRIPTION
**Pull request by Hitarth-Kothari**
Partially works on #21 
need to be linked with the recipe fragment

I made changes to the ingredient class, please review.

### CODE REVIEW

File: code/app/src/main/java/com/example/snackoverflow/AddRecipe.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
+package com.example.snackoverflow;
```
Comment (gokul887) [inline code/app/src/main/java/com/example/snackoverflow/AddRecipe.java:1]: Why is this file a duplicate of [Nagham's pr](https://github.com/CMPUT301F22T22/SnackOverflow/pull/49)

File: code/app/src/main/java/com/example/snackoverflow/Ingredient.java [hunk 2]
```diff
@@ -27,6 +27,14 @@ public Ingredient(String description, String dt, String location, int amount, in
         this.category = category;
     }
 
+    // For Recipe
+    public Ingredient(String description, Integer amount, Integer unit, String category) {
```
Comment (gokul887) [inline code/app/src/main/java/com/example/snackoverflow/Ingredient.java:31]: @raquibk @Bhavnoor-Kaur double check this pls

File: code/app/src/main/java/com/example/snackoverflow/MainActivity.java [hunk 3]
```diff
@@ -31,13 +32,22 @@ protected void onCreate(Bundle savedInstanceState) {
...
 
-        //Test Data
+        //Recipe test
         for (int i =0;i<recipestitle.length;i++){
             recipeDataList.add(new Recipe(recipestitle[i], LocalTime.now(),2.0f,"Lunch","HAHA",new ArrayList<String>(Arrays.asList(new String[]{"Nidal","Nasemm"}))));
         }
-
         recipeArrayAdapter = new RecipeAdapter(this,recipeDataList);
         recipeList.setAdapter(recipeArrayAdapter);
+        Button recipeadd = findViewById(R.id.storage_activity2);
```
Comment (gokul887) [inline code/app/src/main/java/com/example/snackoverflow/MainActivity.java:41]: Look at my other comment about a duplicate

File: code/app/src/main/java/com/example/snackoverflow/AddRecipe.java (hunk 1 above)
Comment (Hitarth-Kothari) [reply code/app/src/main/java/com/example/snackoverflow/AddRecipe.java:1]: ah shit i didn't know the pr would also merge, I had to merge with their branch to make the fragment work

**Comment by gokul887** [comment]
Most of this pr is a possible duplicate with [PR](https://github.com/CMPUT301F22T22/SnackOverflow/pull/49). Only include your functionality

File: code/app/src/main/java/com/example/snackoverflow/MainActivity.java (hunk 3 above)
Comment (Hitarth-Kothari) [reply code/app/src/main/java/com/example/snackoverflow/MainActivity.java:41]: i made changes to her files to make a test for recipe

File: code/app/src/main/java/com/example/snackoverflow/Ingredient.java (hunk 2 above)
Comment (Bhavnoor-Kaur) [reply code/app/src/main/java/com/example/snackoverflow/Ingredient.java:31]: This constructor is missing attribute. Why do we need another constructor? And I noticed, this is the old version of the file! @Hitarth-Kothari Please update your branch with master before making changes!

File: code/app/src/main/java/com/example/snackoverflow/add_ingredient_fragment.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
+package com.example.snackoverflow;
```
Comment (Bhavnoor-Kaur) [inline code/app/src/main/java/com/example/snackoverflow/add_ingredient_fragment.java:1]: This is a duplicate of the fragment @raquibk created. I suggest seeing if we can reuse the fragment which is already created (it makes the code more organized and less complicated). If this absolutely can't be reused, please change the name of the file.

**_Follow naming Convention as in the other files._**

File: code/app/src/main/java/com/example/snackoverflow/add_ingredient_fragment.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+                    + " must implement OnFragmentInteractionListener");
+        }
+    }
+
+    @NonNull
+    @Override
+    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState){
+        //Inflate the layout
+        View view = LayoutInflater.from(getActivity()).inflate(R.layout.reciepe_ingredients_fragment_layout, null);
+        text_input_description = view.findViewById(R.id.text_input_description);
```
Comment (Bhavnoor-Kaur) [inline code/app/src/main/java/com/example/snackoverflow/add_ingredient_fragment.java:62]: Follow variable naming convention to be consistent with others.  Use Camel Case!

Ex: variableName

File: code/app/src/main/res/layout/activity_main.xml [hunk 6]
```diff
@@ -19,4 +19,12 @@
         android:layout_marginTop="600dp"
         android:layout_marginLeft="150dp"/>
 
+    <Button
+        android:id="@+id/storage_activity2"
```
Comment (Bhavnoor-Kaur) [inline code/app/src/main/res/layout/activity_main.xml:23]: Use a more intuitive id to depict what the button is for!

**Review by Bhavnoor-Kaur — CHANGES_REQUESTED** [review]

File: code/app/src/main/res/layout/activity_main.xml (hunk 6 above)
Comment (Hitarth-Kothari) [reply code/app/src/main/res/layout/activity_main.xml:23]: this is just for testing, not for use

**Pull request closed without merging**